deploy/lib/server_config.rb

```diff
@@ -1357,6 +1374,28 @@ def settings
 
 private
 
+  def filter_to_newer_files(files, options)
+    logger.info "Filtering to files which are newer locally than on the server"
+
+    files.select { |file_uri|
+      target_uri = xcc.build_target_uri(file_uri, options)
+
+      q = %Q{"" || xdmp:timestamp-to-wallclock(xdmp:document-timestamp("#{target_uri}"))}
+      result = execute_query q, :db_name => @properties["ml.content-db"]
```

Comment:
Wondering if it would be more efficient to do a single query for all uris (returned as some hash?), or perhaps calls in batches, instead of calling ML for each uri separately. Ideally it would be that fast that it becomes worth keeping it enabled permanently..